src/components/PostData/PostInfo.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useContext } from 'react';
+import classNames from 'classnames';
+import { Post } from '../../types/Post';
+import {
+  ActionTypes,
+  DispatchContext,
+  StateContext,
+} from '../../reducer/store';
+
+type Props = {
+  post: Post
+};
+
+export const PostInfo: React.FC<Props> = ({ post }) => {
+  const dispatch = useContext(DispatchContext);
+  const { selectedPost } = useContext(StateContext);
+  const isPostChosen = selectedPost?.id === post.id;
+
+  const onClickHandler = useCallback(() => {
```

Comment:
```suggestion
  const hancleClick = useCallback(() => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364